plugin/lighthouse/handler_test.go

```diff
@@ -0,0 +1,92 @@
+package lighthouse
+
+import (
+	"context"
+	"os"
+
+	"github.com/coredns/coredns/plugin/pkg/dnstest"
+	"github.com/coredns/coredns/plugin/test"
+	"github.com/miekg/dns"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("[setup] Test basic DNS Handler", func() {
+	Describe("function serveDNS", testDnsServer)
+})
+
+func testDnsServer() {
+	var (
+		lh  *Lighthouse
+		ctx context.Context
+		tc  test.Case
+	)
+	tests := []test.Case{
+		{
+			// service is present.
+			Qname: "mysvc.default.svc.cluster.local.", Qtype: dns.TypeA,
```

Comment:
What happens when the cluster’s suffix isn’t `cluster.local`? See our e2e deployments for an example.

Reply:
It doesn't matter, at least not yet. I have another patch in works that will require configuration of domains we support and that is when we will add more checks.